src/patternfly/components/NumberInput/number-input.scss

```diff
@@ -7,12 +7,24 @@
 
   // form control
   --pf-c-number-input--c-form-control--width-base: calc(var(--pf-global--spacer--sm) * 2 + var(--pf-global--BorderWidth--sm) * 2); // element's padding
+  --pf-c-number-input--c-form-control--width-icon: 0;
+  --pf-c-number-input--m-has-icon--c-form-control--width-icon: var(--pf-global--spacer--xl); // extra width to accommodate a status icon
   --pf-c-number-input--c-form-control--width-chars: 4;
-  --pf-c-number-input--c-form-control--Width: calc(var(--pf-c-number-input--c-form-control--width-base) + var(--pf-c-number-input--c-form-control--width-chars) * 1ch);
+  --pf-c-number-input--c-form-control--Width:
+    calc(
+      calc(
+        var(--pf-c-number-input--c-form-control--width-base) + var(--pf-c-number-input--c-form-control--width-chars) * 1ch
+      ) + var(--pf-c-number-input--c-form-control--width-icon)
+    );
 
   display: inline-flex;
   align-items: center;
 
+  // This modifier can be removed in a breaking change and the extra space created can always be included.
+  &.pf-m-has-icon {
```

Comment:
I would probably either call this icon or status. I looked and we don't have any other classes with `pf-m-[has/is]`.